sbnobj/Common/Calibration/TrackCaloSkimmerObj.h

```diff
@@ -98,6 +101,39 @@ namespace sbn {
      iproc(-1) {}
   };
 
+  struct TrueHit {
+    int16_t cryo; //!< Cryostat of hit
+    int16_t tpc; //!< TPC of hit
+    int16_t plane; //!< Plane of hit
+    int wire; //!< Wire of hit
+    int channel; //!< Channel of hit
+
+    unsigned ndep; //!< Number of depsotions in hit
+    float nelec; //!< Number of electrons in hit
+    float e; //!< energy in hit
+    float pitch; //!< Track pitch for hit
+    float rr; //!< Track residual range for hit
+    int itraj; //!< Index of hit along trajectory
+    Vector3D p; //!< Location of hit
+    float time; //!< Time of hit [ticks]
```

Comment:
I _recommend_ that this class be made a "plain old data" structure by removing the explicit default constructor and initializing the values inline ([`[CF.155]`](https://sbnsoftware.github.io/sbn/codingconv/CodingConventionsExplained.html#CF155)):
```suggestion
  struct TrueHit {
    int16_t cryo { int16_t(-1) }; //!< Cryostat of hit
    int16_t tpc { int16_t(-1) }; //!< TPC of hit
    int16_t plane { int16_t(-1) }; //!< Plane of hit
    int wire { -1 }; //!< Wire of hit
    int channel { -1 }; //!< Channel of hit

    unsigned ndep { 0 }; //!< Number of depsotions in hit
    float nelec { 0.0 }; //!< Number of electrons in hit
    float e { 0.0 }; //!< energy in hit
    float pitch { 0.0 }; //!< Track pitch for hit
    float rr { 0.0 }; //!< Track residual range for hit
    int itraj { -1 }; //!< Index of hit along trajectory
    Vector3D p; //!< Location of hit
    float time { 0.0 }; //!< Time of hit [ticks]
```
Besides the advantages at language level, in this way you don't miss any initialization when you add an element.
I also _recommend_ the same treatment for `Vector3D`:
```cpp
  struct Vector3D {
    float x { std::numeric_limits<float>::signaling_NaN() };
    float y { std::numeric_limits<float>::signaling_NaN() };
    float z { std::numeric_limits<float>::signaling_NaN() };
  };
```
which will allow the above initialization to be written as `Vector3D p { 0.0, 0.0, 0.0 };` (that is one noticeable advantage of having no constructor specified). Otherwise, you will still need a constructor like
```cpp
    TrueHit()
      {
        // set the location to 0.
        p.x = 0;
        p.y = 0;
        p.z = 0;
      }
```

Reply:
So -- I disagree with this convention. Maybe we could discuss at an Analysis Framework meeting.

This is using a C++ feature which is pretty unknown to most people, and most existing data product involve an explicit constructor. 

I can understand that there are some benefits, but I think it should be left up to the developer as to whether they are necessary.

Reply:
This issue is important enough for maintainability for me to insist on the point.
Let's discuss at a meeting then.